webapp/app.py

```diff
@@ -69,21 +67,93 @@ def results():
     return flask.render_template("careers/results.html", **context)
 
 
-@app.route("/careers/admin", methods=["GET", "POST"])
-@app.route("/careers/all", methods=["GET", "POST"])
-@app.route("/careers/commercial-ops", methods=["GET", "POST"])
-@app.route("/careers/design", methods=["GET", "POST"])
-@app.route("/careers/engineering", methods=["GET", "POST"])
-@app.route("/careers/finance", methods=["GET", "POST"])
-@app.route("/careers/hr", methods=["GET", "POST"])
-@app.route("/careers/legal", methods=["GET", "POST"])
-@app.route("/careers/marketing", methods=["GET", "POST"])
-@app.route("/careers/project-management", methods=["GET", "POST"])
-@app.route("/careers/sales", methods=["GET", "POST"])
-@app.route("/careers/tech-ops", methods=["GET", "POST"])
-def department_group():
-    department = flask.request.path.split("/")[2]
-    vacancies = greenhouse_api.get_vacancies(department)
+# Class that collects department-specific content
+class Department(object):
+    __careers_directory = "./templates/careers"
+
+    def __parse_feed_department(feed_department):
+        field = {
+            "cloud engineering": "engineering",
+            "device engineering": "engineering",
+            "web and design": "design",
+            "web & design": "design",
+            "operations": "commercial-ops",
+            "human resources": "hr",
+            "techops": "tech-ops",
+            "product": "product management"
```

Comment:
Are these still required? I was under the impression that the data source has been aligned?

Reply:
Some of these mappings are redundant but there are still a few anomalies; for instance the template corresponding to `web & design` is called `design.html`

Reply:
Amendment to my last comment: the Department container class is also used for comparing the `department` property of vacancies to the external departments list, so all these key:value pairs are necessary

Reply:
Ok, just checking.

Reply:
We want to remove as much logic as possible from this middleware to reduce errors in the future. If that needs URLs to change and adding redirects that would be fine.